dataprofiler/tests/data_readers/test_csv_data.py

```diff
@@ -473,14 +316,23 @@ def test_header_check_files(self):
 
         # add some more files to the list to test the header detection
         # these files have some first lines which are not the header
-        input_file_names = self.input_file_names
-        for input_file in input_file_names:
-            file_encoding = data_utils.detect_file_encoding(input_file['path'])
-            with open(input_file['path'], encoding=file_encoding) as csvfile:
+        for input_file in self.input_file_names:
+            with open(input_file['path'], encoding=input_file['encoding']) as csvfile:
                 data_as_str = ''.join(list(islice(csvfile, 5)))
             header_line = CSVData._guess_header_row(data_as_str, input_file['delimiter'])
             self.assertIn(header_line, input_file['has_header'], input_file['path'])
 
+        for input_buf in self.buffer_list:
+            print(input_buf['path'].read(30))
+            input_buf['path'].seek(0)
```

Comment:
if we add a setup before each test which loops through self._buffer_list seeking to 0, we shouldn't need this.
```
@classmethod
def setUp(cls):
    for buffer in self.buffer_list:
        buffer.seek(0)
```

Reply:
seek(0) isn't required usually, i think i just had that to test.

Reply:
I believe it will be once it is added to all the other tests